buildkitd/entrypoint.sh

```diff
@@ -32,8 +32,16 @@ if [ "$EARTHLY_RESET_TMP_DIR" = "true" ]; then
 fi
 
 if ! lsmod | grep -wq "^ip_tables"; then
-  echo "Legacy ip_tables is not loaded. Switching to iptables-nft."
-  ln -sf /sbin/iptables-nft /sbin/iptables
+  echo "Legacy ip_tables is not loaded."
+
+  if ! uname -a | grep -wiq "microsoft"; then
```

Comment:
Is it possible to use `$OSTYPE` here instead? or if that doesn't work, can we create our own OS detection and move it higher up then just have a `if [ "$OS" = "windows" ]; then` here?

That way we'll keep it fairly consistent in this script if it ever has to grow.

Reply:
This might do the trick:

```suggestion
  if [ "$OSTYPE" = "msys" ]; then
```

Reply:
Does `$OSTYPE` work inside a Docker container? This is more or less from Microsoft on how to detect WSL.

Reply:
I think you might have to do a:

```
GUEST_OS="unknown"

if ! uname -a | grep -wiq "microsoft"; then
    GUEST_OS="windows"
fi
```

then just check 

```
if [ "$GUEST_OS" = "windows" ]; then
```

Reply:
So just lift the OS detection up out of the if? Sure why not.